packages/iota/src/validation/outputs/common.ts

```diff
@@ -0,0 +1,347 @@
+    "Sender Feature", 
+    "Issuer Feature",
+    "Metadata Feature",
+    "Tag Feature"
+];
+
+/**
+ * Map Output type to name.
+ */
+const OUTPUT_TYPE_NAMES = new Map()
+    .set(3, "Basic")
+    .set(4, "Alias")
+    .set(5, "Foundry")
+    .set(6, "Nft");
+
+/**
+ * Validate the output common rules.
+ * @param output The output to validate.
+ * @param protocolInfo The Protocol Info.
+ * @returns The validation result.
+ */
+export function validateCommonRules(output: OutputTypes, protocolInfo: INodeInfoProtocol): IValidationResult {
+    const results: IValidationResult[] = [];
+    const outputType = output.type;
+    const outputName = OUTPUT_TYPE_NAMES.get(output.type);
+
+    results.push(validateAmountIsGreaterThanZero(output.amount, outputName));
+
+    results.push(validateAmountIsLesserThanMaxSupply(output.amount, protocolInfo.tokenSupply, outputName));
+    
+    if (outputType !== TREASURY_OUTPUT_TYPE) {
+        results.push(validateNativeTokens(output.nativeTokens));
+
+        if (output.unlockConditions) {
+            const max = OUTPUT_TYPE_TO_UNLOCK_CONDITIONS.get(outputType).length;
+            const min = OUTPUT_TYPE_TO_MIN_UNLOCK_CONDITIONS.get(outputType);
+            results.push(validateCount(outputName, "Unlock Conditions", output.unlockConditions.length, max, min));
+
+            results.push(validateUnlockConditionsDefineAllowedTypes(output.unlockConditions, outputType, outputName));
+        }
+
+        if (output.features) {
+            const max = OUTPUT_TYPE_TO_FEATURES.get(outputType).length;
+            results.push(validateCount(outputName, "Features", output.features.length, max));
+
+            results.push(validateFeaturesDefineAllowedTypes(output.features, outputType, outputName));
+        }
+
+        if (outputType !== BASIC_OUTPUT_TYPE && output.immutableFeatures) {
+            const max = OUTPUT_TYPE_TO_IMMUTABLE_FEATURES.get(outputType).length;
+            results.push(validateCount(outputName, "Immutable Features", output.immutableFeatures.length, max));
+
+            results.push(validateImmutableFeaturesDefineAllowedTypes(output.immutableFeatures, outputType, outputName));
+        }
+    }
+
+    return mergeValidationResults(...results);
+}
+
+/**
+ * Validate the amount is greater than zero.
+ * @param amount The amount to validate.
+ * @returns The validation result.
+ */
+function validateAmountIsGreaterThanZero(amount: string, outputName: string): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (bigInt(amount).leq(bigInt.zero)) {
+        result = failValidation(result, `${outputName} output amount field must be larger than zero.`);
+    }
+
+    return result;
+}
+
+/**
+ * Validate the amount is lesser than maximum token supply.
+ * @param amount The amount to validate.
+ * @returns The validation result.
+ */
+function validateAmountIsLesserThanMaxSupply(amount: string, tokenSupply: string, outputName: string): IValidationResult {
```

Comment:
Hint! When lines get too long, please use this format:
```
function validateAmountIsLesserThanMaxSupply(
    amount: string,
    tokenSupply: string,
    outputName: string
): IValidationResult {
```
or at least this
```
function validateAmountIsLesserThanMaxSupply(
    amount: string, tokenSupply: string, outputName: string
): IValidationResult {
```
Much easier to read/add/delete parameters, as opposite to just slapping one newline anywhere in between the line.